File: 백준/Silver/1049. 기타줄/기타줄.c
#include <stdio.h>
int main(void) {
    int n, m, pack, each, min_p = 1000, min_e = 1000;
    scanf("%d %d", &n, &m);
    for (int i = 0; i < m; i++) {
        scanf("%d %d", &pack, &each);
        if (min_p > pack) min_p = pack;
        if (min_e > each) min_e = each;
    }
    if (min_p >= min_e * 6)
        printf("%d", min_e * n);
    else
        printf("%d", min_p * (n / 6) + (min_p < min_e * (n % 6) ? min_p : min_e * (n % 6)));
    return 0;
}

File: 백준/Silver/1049. 기타줄/기타줄.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,m=map(int,input().split())
p_list,e_list=[],[]
for _ in range(m):
    pack,each=map(int,input().split())
    p_list.append(pack)
    e_list.append(each)
min_p,min_e=min(p_list),min(e_list)
if min_p>=min_e*6:
    print(min_e*n)
else:
    if min_p<min_e*(n%6):
        print(min_p*(n//6+1))
    else:
        print(min_p*(n//6)+min_e*(n%6))